### PR TITLE
Improve thread safety when logging

### PR DESCRIPTION
**Pull request by nkoenig**
This just makes sure that the event stream is correct in the case of multiple simultaneous events.

Signed-off-by: Nate Koenig <nate@openrobotics.org>

### CODE REVIEW

File: subt_ign/src/GameLogicPlugin.cc [hunk 1]
```diff
@@ -1189,13 +1214,15 @@ void GameLogicPluginPrivate::Finish()
     this->logStream.flush();
     this->eventStream.flush();
```
Comment (iche033) [inline subt_ign/src/GameLogicPlugin.cc:1215]: I think this flush call can be moved to `LogEvent` function

File: subt_ign/src/GameLogicPlugin.cc [hunk 2]
```diff
@@ -1375,6 +1402,13 @@ bool GameLogicPluginPrivate::OnPoseFromArtifact(
   return result;
 }
 
+/////////////////////////////////////////////////
+void GameLogicPluginPrivate::LogEvent(const std::string &_event)
+{
+  std::lock_guard<std::mutex> lock(this->eventMutex);
+  this->eventStream << _event;
```
Comment (iche033) [inline subt_ign/src/GameLogicPlugin.cc:1409]: add `this->eventStream.flush();` here?

File: subt_ign/src/GameLogicPlugin.cc (hunk 1 above)
Comment (nkoenig) [reply subt_ign/src/GameLogicPlugin.cc]: Removed: 762e1dd713a9e608ed1e4b302deb8273df15fc1d

File: subt_ign/src/GameLogicPlugin.cc (hunk 2 above)
Comment (nkoenig) [reply subt_ign/src/GameLogicPlugin.cc:1408]: Added: 762e1dd713a9e608ed1e4b302deb8273df15fc1d

**Pull request merged**